### ISSUE TITLE
Validate typealias naming

### ISSUE THREAD

**Issue by jpsim** [realm/SwiftLint#49]
Sort of depends on rdar://18845613 getting fixed.

**Comment by jpsim** [comment]
I don't know why I said this depended on rdar://18845613, that's an unrelated radar.

**Comment by masters3d** [comment]
To be uppercase? any other validations?

**Comment by jpsim** [comment]
same naming rules as types.

**Comment by marcelofabri** [comment]
I think `TypeNameRule` covers this, doesn't it?

**Comment by jpsim** [comment]
No.

```shell
$ echo "typealias myType = Void" | swiftlint lint --use-stdin 
Done linting! Found 0 violations, 0 serious in 1 file.
```

**Comment by marcelofabri** [comment]
Apparently, SourceKit doesn't provide info about `typealias` so we'd need to do that manually:

```bash
$ sourcekitten structure --text "typealias myType = Void"  
{
  "key.offset" : 0,
  "key.diagnostic_stage" : "source.diagnostic.stage.swift.parse",
  "key.length" : 23
}


$ sourcekitten syntax --text "typealias myType = Void"    
[
  {
    "offset" : 0,
    "length" : 9,
    "type" : "source.lang.swift.syntaxtype.keyword"
  },
  {
    "offset" : 10,
    "length" : 6,
    "type" : "source.lang.swift.syntaxtype.identifier"
  },
  {
    "offset" : 19,
    "length" : 4,
    "type" : "source.lang.swift.syntaxtype.typeidentifier"
  }
]
```
